김현국/app/src/main/java/com/example/myapplication/ui/adapter/SearchAdapter.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.myapplication.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.myapplication.databinding.ItemSearchBinding
+import com.example.myapplication.ui.model.PresenterRepository
+
+/**
+ * @author 김현국
+ * @created 2022/04/25
+ */
+class SearchAdapter(
+    onItemClick: OnItemClick
+) :
+    ListAdapter<PresenterRepository, SearchAdapter.RepoViewHolder>(diffUtil) {
+
+    interface OnItemClick {
+        fun onItemClick(repository: PresenterRepository)
+    }
+    var onItemClick: OnItemClick = onItemClick
+
+    companion object {
+        private val diffUtil = object : DiffUtil.ItemCallback<PresenterRepository>() {
+            override fun areItemsTheSame(
+                oldItem: PresenterRepository,
+                newItem: PresenterRepository
+            ): Boolean {
+                return oldItem.id == newItem.id
+            }
+
+            override fun areContentsTheSame(
+                oldItem: PresenterRepository,
+                newItem: PresenterRepository
+            ): Boolean {
+                return oldItem == newItem
+            }
+        }
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): RepoViewHolder {
+
+        return RepoViewHolder(
+            ItemSearchBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: RepoViewHolder, position: Int) {
+        holder.bind(getItem(position))
+        holder.itemView.setOnClickListener {
+            onItemClick.onItemClick(getItem(position))
+        }
+    }
+
+    class RepoViewHolder(private val binding: ItemSearchBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(repo: PresenterRepository) {
+            binding.tvMainName.text = repo.name
+            if (repo.language != null) {
+                binding.tvMainLanguage.text = repo.language
+            }
```

Comment:
```suggestion
repo?.language?.let { binding.tvMainLanguage.text = it } 
```
과 같이도 쓸 수 있을 것 같은데, 어디선가 let함수 요샌 또 안쓰기도 한다는걸 본것 같기도 해서 다른분들도 새로운 의견 주실 것 같아서 일단 적어봅니다

Reply:
showToast(result.message ?: "예상치 못한 오류 발생") 이런 작성은 어떤가요?

Reply:
그것도 좋아보이네요!